cranelift/codegen/src/ir/exception_table.rs

```diff
@@ -156,11 +177,32 @@ impl ExceptionTableData {
         self.sig
     }
 
+    /// Get a mutable handle to this exception table's signature.
+    pub(crate) fn signature_mut(&mut self) -> &mut SigRef {
+        &mut self.sig
+    }
+
     /// Clears all entries in this exception table, but leaves the function signature.
     pub fn clear(&mut self) {
         self.tags.clear();
         self.targets.clear();
     }
+
+    /// Push a catch target onto this exception table.
+    pub fn push_catch(&mut self, tag: Option<ExceptionTag>, block_call: BlockCall) {
+        self.tags.push(tag.into());
+
+        // We don't want to push the target, because the normal return is always
+        // last. Instead we want to insert it in the second to last spot.
+        //
+        // It is always true that `self.tags.len() == self.targets.len() + 1`
+        // because of the presence of the normal return which does not have an
+        // associated tag *except* for the case where we have cleared the
+        // exception table, in which case both vecs will be empty. We handle the
+        // cleared case by using saturating subtraction here.
```

Comment:
This feels like a somewhat error-prone API -- it imposes an ordering requirement (or rather, the first `push_catch` is not really a catch but the normal return). Perhaps we could have a separate setter for the normal return entry starting from an empty exception table, and then assert here that the normal return is present?